### PR TITLE
feat: coppy and open address

### PR DESCRIPTION
**Pull request by petia-latinova**

### CODE REVIEW

File: app/src/components/UserAddress.vue [hunk 1]
```diff
@@ -2,15 +2,25 @@
...
 </template>
 
 <script>
+import ethersConnect from '@/store/modules/ethers/ethersConnect';
+
 export default {
   name: 'UserAddress',
+  data() {
+    return {
+      ethersConnect: ethersConnect,
```
Comment (imollov) [inline app/src/components/UserAddress.vue]: Objects inside data option are made reactive by Vue, which is not necessary. You can have a computed property for the explorer URL instead.

File: app/src/components/UserAddress.vue [hunk 2]
```diff
@@ -20,6 +30,21 @@ export default {
...
   },
+  methods: {
+    copyAddress() {
+      const clipboardData =
+        event.clipboardData ||
+        window.clipboardData ||
+        event.originalEvent?.clipboardData ||
+        navigator.clipboard;
+
+      clipboardData.writeText(this.address);
```
Comment (imollov) [inline app/src/components/UserAddress.vue]: We need a more reliable approach. You can also check https://www.npmjs.com/package/v-clipboard

File: app/src/store/modules/ethers/ethersConnect.js [hunk 3]
```diff
@@ -190,7 +214,7 @@ export async function connect() {
       // EIP-1193 userRejectedRequest error
       // If this happens, the user rejected the connection request.
       event.$emit(EVENT_CHANNEL, MSGS.NOT_READY, err);
-    } else {
+    } else {
```
Comment (imollov) [inline app/src/store/modules/ethers/ethersConnect.js]:
```suggestion
    } else {
```

File: app/src/store/modules/ethers/ethersConnect.js [hunk 4]
```diff
@@ -73,6 +73,30 @@ export async function getNetName() {
   }
 }
 
+// get the explorer url of this network
+export function getNetExplorerUrl() {
+  switch (chainId) {
+    case '0x1':
+      return 'https://etherscan.io/';
+    case '0x2':
```
Comment (imollov) [inline app/src/store/modules/ethers/ethersConnect.js:80]: Same as default.

File: app/src/store/modules/ethers/ethersConnect.js [hunk 5]
```diff
@@ -73,6 +73,30 @@ export async function getNetName() {
...
+    case '0x3':
+      return 'https://ropsten.etherscan.io/';
+    case '0x4':
+      return 'https://rinkeby.etherscan.io/';
+    case '0x5':
+      return 'https://goerli.etherscan.io/';
+    case '0x2a':
+      return 'https://kovan.etherscan.io/';
+    case undefined:
+    case null:
```
Comment (imollov) [inline app/src/store/modules/ethers/ethersConnect.js]: Same as default.

File: app/src/store/modules/ethers/ethersConnect.js [hunk 6]
```diff
@@ -73,6 +73,30 @@ export async function getNetName() {
   }
 }
 
+// get the explorer url of this network
+export function getNetExplorerUrl() {
```
Comment (imollov) [inline app/src/store/modules/ethers/ethersConnect.js]:
```suggestion
export function getExplorerUrl() {
```

File: app/src/components/UserAddress.vue [hunk 7]
```diff
@@ -2,15 +2,25 @@
   <b-tag rounded class="user-address">
     <span class="address" :title="address">
       <img :src="avatarUrl" class="img mr-1" />
-      <template v-if="truncate">{{ address | truncate(10) }}</template>
-      <template v-else>{{ address }}</template>
+      <a target="_blank" :href="ethersConnect.getNetExplorerUrl() + 'address/' + address">
+        <template v-if="truncate">{{ address | truncate(10) }}</template>
+        <template v-else>{{ address }}</template>
+      </a>
+      <span @click="copyAddress"><b-icon icon="clipboard" type="is-primary" /></span>
```
Comment (imollov) [inline app/src/components/UserAddress.vue]: Icon is not aligned and spaced properly. It also needs to be sized and colored differently. Additionally it needs to have cursor pointer.

**Comment by imollov** [comment]
Please split the commit into two separate ones for each issue. Also, add the issues you're addressing in the description.

File: app/src/components/UserAddress.vue (hunk 7 above)
Comment (petia-latinova) [reply app/src/components/UserAddress.vue]: Where and to what will be properly aligned and spaced and what is the right size and color? The color needs to be the reverted version of the icon?

File: app/src/store/modules/ethers/ethersConnect.js (hunk 4 above)
Comment (imollov) [reply app/src/store/modules/ethers/ethersConnect.js:80]: Actually `undefined` and `null` should throw error or return empty string. They can be omitted though if you rely on default to catch them.

File: app/src/components/UserAddress.vue (hunk 7 above)
Comment (imollov) [reply app/src/components/UserAddress.vue]: looks good now

**Comment by imollov** [comment]
Please execute `npm run lint` and commit again.

**Comment by imollov** [comment]
> Please split the commit into two separate ones for each issue. Also, add the issues you're addressing in the description.

This is still pending.

File: app/src/main.js [hunk 8]
```diff
@@ -5,7 +5,8 @@ import App from './App.vue';
 import router from './router';
 import store from './store';
 import './main.scss';
-
+import Clipboard from 'v-clipboard'
```
Comment (imollov) [inline app/src/main.js]: External packages on top.

**Review by imollov — APPROVED** [review]

**Pull request merged**